concepts/contexts.py

```diff
@@ -711,6 +711,25 @@ def shape(self) -> _common.Shape:
             Shape(objects=6, properties=10)
         """
         return _common.Shape._from_pair(self.objects, self.properties)
+    
+    @tools.lazyproperty
+    def density(self) -> float:
+        """The density of the context.
+
+        Returns:
+            Context density.
+        
+        Example:
+            >>> import concepts
+            >>> context = concepts.Context(['cat', 'dog', 'bird'],
+            ...                   ['barks', 'meows', 'chirps', '4 legs', '2 legs'],
+            ...                   [(False, True, False, True, False),
+            ...                    (True, False, False, True, False),
+            ...                    (False, False, True, False, True)])  # doctest: +ELLIPSIS
+            >>> context.density
+            0.4
+        """
+        return sum(map(sum, self.bools)) / (self.shape.rows * self.shape.columns)
```

Comment:
> `.bools()`

No need to transform into bools first: `sum(intent.count() for i in ccontext._intents)`.

> `self.shape.rows * self.shape.columns`

`objects`, `properties` (cf. rest of the module)